modules/grids/spec/components/grids/widgets/subitems_spec.rb

```diff
@@ -38,27 +38,53 @@ def render_component(...)
   end
 
   let(:project) { build_stubbed(:project) }
-  let(:user) { build_stubbed(:user) }
   let(:params) { {} }
 
   current_user { user }
 
-  subject(:rendered_component) do
-    render_component(project, current_user:, **params)
+  subject(:rendered_component) { render_component(project, current_user:, **params) }
+
+  shared_examples "empty-state without action" do
+    it "renders empty blankslate without action button" do
+      expect(rendered_component).to have_test_selector(empty_selector)
+      expect(rendered_component).to have_text(empty_message)
+      expect(rendered_component).to have_no_css(".blankslate-action")
+    end
+  end
+
+  shared_examples "empty-state with action" do
+    it "renders empty blankslate with action button" do
+      expect(rendered_component).to have_test_selector("subitems-widget-empty")
+      expect(rendered_component).to have_text("This widget is currently empty.")
+      expect(rendered_component).to have_css(".blankslate-action")
+    end
   end
 
   context "with no children" do
-    it "renders a message" do
-      expect(rendered_component).to have_primer_text "There are no visible children.", color: "subtle"
+    let(:user) { build_stubbed(:admin) }
+    let(:empty_selector) { "subitems-widget-empty" }
+    let(:empty_message)  { "This widget is currently empty." }
+
+    it_behaves_like "empty-state with action"
+
+    context "when user cannot add subprojects but can view" do
+      let(:user) { build_stubbed(:user) }
+
+      before do
+        allow(user).to receive(:allowed_in_project?).with(:view_project, project).and_return(true)
+        allow(user).to receive(:allowed_in_project?).with(:add_subprojects, project).and_return(false)
```

Comment:
Here it already bite you. This test is true, although the behaviour is not working as intended. You need to create two projects in the test. In the parent, the user is member of and allowed to see, but the child the user is not allowed to see. Thus you can actually test what you want to test